modules/ve-cx/init/ve.init.mw.CXTarget.js

```diff
@@ -1110,11 +1118,24 @@ ve.init.mw.CXTarget.prototype.changeContentSource = function (
  *
  * @param {number} sectionNumber
  */
-ve.init.mw.CXTarget.prototype.prefetchTranslationForSection = function ( sectionNumber ) {
+ve.init.mw.CXTarget.prototype.prefetchTranslationForSection = function ( sectionNumber, set_it ) {
 	const $section = this.sourceSurface.$element.find( '#cxSourceSection' + sectionNumber );
 	if ( $section.length ) {
 		this.MTManager.getPreferredProvider().then( function ( provider ) {
-			this.translateSection( $section.prop( 'id' ), provider );
+			if (!set_it) {
+				this.translateSection( $section.prop( 'id' ), provider );
+			} else	 {
+				// TODO:
+				// هذا الجزء لا يعمل بشكل صحيح
+				/*
+					Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'getModel')
+    				at ve.init.mw.CXTarget.setSectionContent (ve.init.mw.CXTarget.js:882:40)
+				*/
+				this.translateSection( $section.prop( 'id' ), provider, false ).then( function ( content ) {
+					ve.init.mw.CXTarget.prototype.setSectionContent( $section, content, provider );
```

Comment:
I believe this could be changed to `this.setSectionContent` instead of calling it from the prototype directly?